### ISSUE TITLE
Allow AWS Prefix Lists for sshAccess and kubernetesApiAccess

### ISSUE THREAD

**Issue by mcbanderson** [kubernetes/kops#12925]
/kind feature

**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

AWS provides [prefix lists](https://docs.aws.amazon.com/vpc/latest/userguide/managed-prefix-lists.html) as a mechanism for managing security group rules and route tables. These prefix lists can be used in security group rules in place of a CIDR and AWS will automatically handle all CIDRs in the list with only one security group rule.

We would like to be able to utilize these prefix lists for [sshAccess](https://github.com/kubernetes/kops/blob/master/docs/cluster_spec.md#sshaccess) and [kubernetesApiAccess](https://github.com/kubernetes/kops/blob/master/docs/cluster_spec.md#kubernetesapiaccess) in our cluster spec to simplify security group management and avoid hitting AWS' hard limit on the number of security group rules.

**2. Feel free to provide a design supporting your feature request.**

From a user interface standpoint we would simply like to see kops allow the prefix list id (name would also be acceptable, but that might be more complicated to implement) in the pertinent parts of the cluster spec and then create the inbound security group rules with the prefix list instead of the CIDR.

```
spec:
  kubernetesApiAccess:
    - 12.34.56.78/32
    - pl-abc123
```

**Comment by rifelpet** [comment]
This sounds reasonable and should be pretty straight forward. If anyone wants to implement this, it should just involve:

* Adding a PrefixList field to the SecurityGroupRule task [here](https://github.com/kubernetes/kops/blob/d0b577183fe9df0f562905d1ff9190262636b38b/upup/pkg/fi/cloudup/awstasks/securitygrouprule.go#L35), adding similar logic throughout that file as the existing CIDR and IPv6CIDR fields.
* Update references to KubernetesAPIAccess ([1](https://github.com/kubernetes/kops/blob/d0b577183fe9df0f562905d1ff9190262636b38b/pkg/model/awsmodel/external_access.go#L151), [2](https://github.com/kubernetes/kops/blob/d0b577183fe9df0f562905d1ff9190262636b38b/pkg/model/awsmodel/api_loadbalancer.go#L358), [3](https://github.com/kubernetes/kops/blob/d0b577183fe9df0f562905d1ff9190262636b38b/pkg/model/awsmodel/api_loadbalancer.go#L407), and similar for `SSHAccess` and `AdminAccess`) to check for a `pl-` prefix on the list element, conditionally using the field in the SecurityGroupRule task's PrefixList field.

**Comment by hierynomus** [comment]
I'll pick this up, expect a PR somewhere next week.

**Comment by hierynomus** [comment]
With the merge of #13113, this issue can be closed I think @hakman

**Comment by hakman** [comment]
> With the merge of #13113, this issue can be closed I think @hakman

I think so. Should be available in kOps 1.23. Thanks again @hierynomus!
